### ISSUE TITLE
Does someone use this project to develop a chatbot/conversation agent?

### ISSUE THREAD

**Issue by loveJasmine** [ufal/neuralmonkey#439]
any experience/idea to share?

**Comment by tomkocmi** [comment]
I have experimented with it in Fall, I have used machine translation config but provided it with data pairs "question, answer". It worked, but nothing fantastic, you would need to play with it more and maybe also defined better metric since default config will most likely be answering regular answers (like "I don't know', "How are you?" etc.). It also heavily depends on the dataset you will use (I played with tv subtitles).

**Comment by loveJasmine** [comment]
https://github.com/ufal/neuralmonkey/issues/215#event-1089158312

maybe we should try this new feature...

**Comment by loveJasmine** [comment]
@tomkocmi 
do you use the 'Byte Pair Encoding' on chatbot?
if you do, how about that?

**Comment by tomkocmi** [comment]
I have used the bpe. It was a really small project, so I copied the Machine Translation config file, which contains bpe, and run it on the subtitles.

**Comment by tomkocmi** [comment]
As for the Self-critical Sequence Training I still didn't read the paper, so I cannot say if it would be useful. If you will be successful with the ChatBot, let me know :)

**Comment by jlibovicky** [comment]
SCST uses REINFORCE to optimize towards a metric, e.g. BLEU score in case of MT. I don't know about any such metric for chat bots, but if there is one, reinforcement learning may help.

**Comment by loveJasmine** [comment]
@jlibovicky 

the task 'chatbot' has the same type of training data ,namely sentence pairs, why not use the BLEU/ROUGE as metrics for  chat bot?

**Comment by tomkocmi** [comment]
It could be interesting to implement this metric, which could fight the problem of too general answers: https://arxiv.org/pdf/1510.03055.pdf

**Comment by loveJasmine** [comment]
@tomkocmi 
I had done the MMI-antilm model in my project, it looks very slowly in decoding although it does diversified the response.

**Comment by tomkocmi** [comment]
Thats great to hear. So can the chatbot follow some conversation or is the diversity still too small?
